tests/basics/int_big_to_small.py

```diff
@@ -5,6 +5,17 @@
     print("SKIP")
     raise SystemExit
 
+# Skip this test on "REPR B" where (1<<29 + 1) is not a small int.
+k = 1 << 29
+micropython.heap_lock()
+try:
+    k = k + 1
+except MemoryError:
+    print("SKIP")
```

Comment:
Instead of SKIP'ing and essentially duplicating this test below for REPR_B, how about using this bit of code here as a feature detection for the integer values to iterate over.

Eg:
```py
limit_value = 1 << 29
micropython.heap_lock()
try:
    k = limit_value + 1
except MemoryError:
    # REPR B, use a smaller limit value
    limit_value = 1 << 28
micropython.heap_unlock()

for d in (0, 27, limit_value, -1861, -limit_value):
    ...
```

Oh... I guess that makes the .exp file different... which is probably too difficult to work around.  In that case I guess it's OK to have separate test files.